pkg/cli/login/loginoptions.go

```diff
@@ -162,16 +162,26 @@ func (o *LoginOptions) getClientConfig() (*restclient.Config, error) {
 	clientConfig.Host = o.Server
 	clientConfig.Insecure = o.InsecureTLS
 
-	if !o.InsecureTLS {
-		// use specified CA or find existing CA
+	if !clientConfig.Insecure {
+		// Try to use the specified CA. Then fall back into searching kubeconfig.
+		// In case no CA is found, still check the kubeconfig for insecure-skip-tls-verify.
 		if len(o.CAFile) > 0 {
 			clientConfig.CAFile = o.CAFile
 			clientConfig.CAData = nil
-		} else if caFile, caData, ok := findExistingClientCA(clientConfig.Host, *o.StartingKubeConfig); ok {
-			clientConfig.CAFile = caFile
-			clientConfig.CAData = caData
+		} else if cluster := findClusterWithCA(clientConfig.Host, *o.StartingKubeConfig); cluster != nil {
+			clientConfig.Insecure = cluster.InsecureSkipTLSVerify
+			clientConfig.CAFile = cluster.CertificateAuthority
+			clientConfig.CAData = cluster.CertificateAuthorityData
+		} else if cluster := findCluster(clientConfig.Host, *o.StartingKubeConfig, nil); cluster != nil {
+			clientConfig.Insecure = cluster.InsecureSkipTLSVerify
 		}
 	}
+	// In case the kubeconfig contained insecure-skip-tls-verify, we must unset the CA settings,
+	// otherwise an error is returned from k8s transport init machinery.
+	if clientConfig.Insecure {
+		clientConfig.CAFile = ""
```

Comment:
We shouldn't do this implicitly, especially without any prompting about the insecure behavior.

Reply:
If we error out before than that, we won't need this.